.test-infra/dataproc/cleanup.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+#    Licensed to the Apache Software Foundation (ASF) under one or more
+#    contributor license agreements.  See the NOTICE file distributed with
+#    this work for additional information regarding copyright ownership.
+#    The ASF licenses this file to You under the Apache License, Version 2.0
+#    (the "License"); you may not use this file except in compliance with
+#    the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+set -Eeuo pipefail
+GCLOUD_REGION="us-central1"
+clustersList=( )
+toDeleteList=( )
+
+
+generatedResources=("beam-loadtests-go-cogbk-flink" "beam-loadtests-python-cogbk-flink" \
```

Comment:
do we know when the cleanup https://github.com/apache/beam/blob/e27d8d6bb744daa098e21fc71e6371ca5f2062cf/.test-infra/jenkins/Flink.groovy#L96 is not working?

Reply:
It looks like it happened in the past due to some errors in the code, the misconfiguration was fixed in [BEAM-12898](https://issues.apache.org/jira/browse/BEAM-12898), other possible scenarios might be the existence of interrupted or aborted jobs.

Reply:
The PostBuild scripts are supposed to run even if the task is failed or aborted, there was a fix previously in https://github.com/apache/beam/commit/a84d9cb285c5d15eb34f4a3126015ae1f1da003f.

I'm not against adding additional clean up script, but do have concern that it is duplicating the clean up steps and overtime this may become outdated more easily than maintaining the PostBuild steps.

Reply:
During the testing process for [BEAM-12898](https://issues.apache.org/jira/browse/BEAM-12898) sometimes when the Jenkins node went completely down due to overloading, PostBuild Scripts were not executed, this situation is rare though, not sure if events like the Jenkins downtime that happened a few days ago would also trigger this.

The computation load of this job is not high , an option could be to execute it less frequently just as a backup in case that something eventually happens.